### ISSUE TITLE
Blazor Theme: User Profile Component and login component do not show full screen

### ISSUE THREAD

**Issue by thabaum** [oqtane/oqtane.framework#3241]
Switch to default blazor theme 
noticed no log in or a way get to the dashboard in full screen desktop. 
Make the window small for mobile and the nav menu toggler shows is displayed and all is displayed but in different areas depending on how far you shrink it as shown below:

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/d04961c4-0f00-45c9-a227-05ecd0b918e5)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/bb303d9c-95bc-4155-b9b7-63e8c4818294)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/1941337d-e72a-4af6-9132-82f1cc68001e)

Not sure where it should go but I am thinking the needs to be in the sidebar if not at the top of content to the right side of the window.

**Comment by sbwalker** [comment]
@thabaum not sure what you are talking about - the screen shots clearly show the login/logout button as well as the cog icon for displaying the Control Panel. The fact that the nav menu hamburger icon changes position is by design - there is responsive design logic to move the icon depending on the screen size.

**Comment by thabaum** [comment]
Sorry for not being clear enough, FULL PAGE the last one, no go...

**Comment by sbwalker** [comment]
I cannot duplicate. I have expanded my full screen display as wide as possible and the login button and cog icon are still visible. What browser are you using?

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/7c629c15-aaed-4f95-90f1-45c5e9b555f8)

**Comment by thabaum** [comment]
I see it is working for you, wonder why not for me.  I will test again on another installation.  Maybe this one I am using is bugged, or something I am adding to it is bugging it out.  But really not much going on in this installation other than I was making a radzen module so I will investigate this further.

**Comment by sbwalker** [comment]
@thabaum if you have injected other CSS references into the site or page it could affect the behavior of the Blazor theme.

**Comment by thabaum** [comment]
```
<RadzenDialog />
    <RadzenNotification />
    <RadzenContextMenu />
    <RadzenTooltip />
```
    
    I have this in Site Settings > Body Content and it loads properly at the bottom.
   
 but removing this had no affect.
 
 I am closing this since it is working for you.
 
 I am just creating a module with Radzen datagrid, chart, dialog and notifications.  And I am going just on the homepage with no module loading other than defaults.

**Comment by sbwalker** [comment]
@thabaum those tags are custom Razor components - they are not valid HTML tags. If you include them in Head Content or Body Content they will be injected into the HTML output but they will not be executed. In order to use custom Razor components they must be specified within a Razor component  which is compiled and deployed to your site.

**Comment by thabaum** [comment]
I found that out the hard way :) 

I am trying to recall where I was at in regards to the loading of resources.

I was going to say I need a way to inject `@using Radzen` and these components into the theme.  This is where it works best for positioning.  Is there a way to do this?

**Comment by sbwalker** [comment]
@thabaum Each third party component library has specific requirements for integration. Please see the following thread for a discussion on the common issues:

https://github.com/oqtane/oqtane.framework/discussions/690

This example for Telerik is the best current example (however Radzen may have its own requirements):

https://github.com/nicpitsch/Oqtane.TelerikModule1

And this is a good resource for Radzen (although it is a bit outdated and probably needs to be updated):

https://blazorhelpwebsite.com/ViewBlogPost/46

**Comment by thabaum** [comment]
I did read over the Telerik module a few days ago i didnt see anything that applies here.  The other tutorial follows pretty much what I have done except I did a few things a few ways differently as well as the way explained there which is nice as it gives another way to make it work.
So I did add it to the _Imports file, however I think when adding to a theme if I recall this did not work.  If just loading into the module itself it works.
I do have it working with default Oqtane theme, but with Blazor theme some reason Radzen does not work.
I must have another issue within my installation.
I have these Radzen components dailog/tooltip/notification/contextmenu working in modules in two different scenarios:
1. adding components and resources to include to the Theme directly
2. adding the same directly in the module component (which seems to shift things around out of center resulting in a lot of CSS fixes fighting responsiveness and so on...)

To fire off Radzen pretty much all I needed to do was install in the client project the package and use this:
```
        new() { ResourceType = ResourceType.Stylesheet, Url = "_content/Radzen.Blazor/css/default-base.css" },
        new() { ResourceType = ResourceType.Script, Url = "_content/Radzen.Blazor/Radzen.Blazor.js" },
```
to the theme default page or directly in each master module component (if it has child components they did not need it) or ThemeInfo.cs/ModuleInfo.cs is suggested but I put them where the others where located (I have a reason for this)

and
```
@using Radzen
@using Radzen.Blazor 
```
in the module `_Imports.cs` file
It is just recommended to put these in the _Host.cshtml main layout which I had best results including into the theme and it works great for out of the box positioning (ie dialog is centered)  

The Z index needs some CSS love to get on top of the Oqtane navbar as radzen does not put it out on a high enough layer number to get on top of all of the oqtanes default theme.
So I have had it working in both adding to a theme (best results) and adding to the module which makes a CSS fix a bit of an out of box pain.

What i would love to do is only inject this on pages that have the module so it is not always loading these resources on every page into the bottom of the body which is why I tried the Body Content with negative results (as you mentioned this does not get rendered when added here).  This way it has a harder time to conflict with other modules UI Libraries is what I am thinking.
It's time to do this all with fresh start again as I may have introduced a bug while browsing around in the code base version I am using for these developments.   I have about 10 modules created using different approaches and features.  I am tempted to putting it out as a public repo but not sure I have the spare time to manage one.

**On another note:  **
I have done some MAUI development using MVVP patterns and native features along with firing up Oqtane to see it at work as a Hybrid Blazor Mobile Maui app, really cool :).  I am hoping to mix the two (native and hybrid) using Oqtane.  Not sure if anyone has or is doing this?  But I am curious how these different UI libraries will look and act when I start testing them with Oqtane Maui project.

To run a Maui project I had to include the project into the solution (not sure if not included intentionally to help limit the projects building time so it can be included on a need bases)  Then fire up the server project from another VS instance so that the API is available, then launch the Maui project in the event anyone wants to try and hasn't yet.
After reviewing those tutorials, I pretty much followed those, The IHostResources is deprecated if I recall from blogs from this tutorial.

[https://github.com/oqtane/oqtane.framework/discussions/2818](https://github.com/oqtane/oqtane.framework/discussions/2818)

It's not like I dont have this working.  This was my first real attempt at creating a fairly complex module in Oqtane Framework where I am pulling data from an API to display in charts/datagrid and creating dialogs with additional information from them when clicked using the Radzen UI library.
I run the project as "Server" however I only installed the Radzen package in "Client" since this is where all the UI exists in Oqtane.  The tutorial for radzen says to install in both, I didnt... it works with just client project package install which would make it easier to maintain updating.

It works this way just fine for me from what I can tell.  From past projects with blazor generally you create both server UI pages and client UI pages, but since Oqtane uses an API regardless of type (Server/WebAssembly/Maui) from the client project it is almost always "Client" UI except with few components for UI located in Server>Pages which maybe some should be moved to client project as well to keep them all together like Login for example.  If I update the Login component to use Radzen for example then I would need to install this package here as well for it to work.  SInce I am creating a module, it dont matter form a UI perspective as long as you keep all UI components in the Client project.

But since I am just currently creating "Oqtane Client Project UI" components for this module which are included in server builds I did not see the need to install these UI packages in the server project as described in blazorhelpwebsite.com blog post.

My background here is I created a few blazor projects from scratch such as eCommerce app and an inventory app... and also with .net MAUI using stripe checkout.  When creating apps for both server and client I had to make pages, services for both, with Client using client services to talk to the server controller to communicate with the server repository which talks to the database context in general.  Oqtane does not do this for most components/pages, it always uses the API controller regardless of server/webassembly type in most all cases I can see relating to it's UI components.  This threw me off a while ago which I made a comment about how I may have done this a bit differently, but with .NET 8 it don't matter... it's almost like you can see the future!

**Bottom line:**
@sbwalker I am going to rebuild this module in a new installation of Oqtane with latest dev branch following all the details I have learned along the way to eliminate any bugs I may have introduced accidentally along with way while experimenting with Radzen UI component approaches.  This module should impress some people maybe if I can pull of what I think I can do as it will do something I have not seen done by a module vendor for Oqtane or any DNN module that I am aware of, nothing super special but still pretty cool for what it should pull off for a feature.  The new developer features recently added to the framework should help pull it all off as the logic will do the rest.
Thank you!
